### PR TITLE
SW-2727 Calculate total accession withdrawn quantities

### PR DESCRIPTION
**Pull request by sgrimm**
Keep track of the total amount withdrawn from each accession, and expose the
totals as search fields.

Reduce a bit of repetition by including the raw quantity and units fields in the
output of `SearchTable.weightFields()`.

Add a migration to add the totals to existing accessions. Since this isn't the
first time we've needed a "go through all the accessions one by one and compute
something" migration, add a helper method to `AccessionStore` to handle the
iteration logic. Once this migration is done, we'll remove
`MigrateWithdrawnTotals` but we can keep the helper method.

### CODE REVIEW

**Comment by sgrimm** [comment]
Current dependencies on/for this PR:
* main
  * **PR #932** <a href="https://app.graphite.dev/github/pr/terraware/terraware-server/932" target="_blank">Graphite</a> 
    * **PR #936** <a href="https://app.graphite.dev/github/pr/terraware/terraware-server/936" target="_blank">Graphite</a> 
    * **PR #937** <a href="https://app.graphite.dev/github/pr/terraware/terraware-server/937" target="_blank">Graphite</a> 
    * **PR #931** <a href="https://app.graphite.dev/github/pr/terraware/terraware-server/931" target="_blank">Graphite</a>  👈
      * **PR #934** <a href="https://app.graphite.dev/github/pr/terraware/terraware-server/934" target="_blank">Graphite</a> 

This comment was auto-generated by [Graphite](https://app.graphite.dev/github/pr/terraware/terraware-server/931?utm_source=stack-comment).

**Comment by sgrimm** [comment]
Oops, this one turned out to be half-baked; converting to draft for now and will update.

File: src/main/kotlin/com/terraformation/backend/seedbank/db/AccessionStore.kt [hunk 1]
```diff
@@ -858,4 +868,43 @@ class AccessionStore(
...
+                .select(ACCESSIONS.ID)
+                .from(ACCESSIONS)
+                .where(condition)
+                .and(nextAccessionId?.let { ACCESSIONS.ID.gt(it) } ?: DSL.trueCondition())
+                .limit(1)
+                .forUpdate()
+                .skipLocked()
+                .fetchOne(ACCESSIONS.ID)
+
+        nextAccessionId?.let { accessionId -> func(fetchOneById(accessionId)) }
```
Comment (karthikbtf) [inline src/main/kotlin/com/terraformation/backend/seedbank/db/AccessionStore.kt:910]: should we be catching exceptions here or leave it to the func?

File: src/main/kotlin/com/terraformation/backend/seedbank/db/MigrateWithdrawnTotals.kt [hunk 2]
```diff
@@ -0,0 +1,66 @@
...
+                DSL.selectOne()
+                    .from(WITHDRAWALS)
+                    .where(WITHDRAWALS.ACCESSION_ID.eq(ACCESSIONS.ID))
+                    .and(WITHDRAWALS.WITHDRAWN_QUANTITY.isNotNull))
+
+    systemUser.run {
+      accessionStore.forEachAccession(condition) {
+        val accession = it.withCalculatedValues()
+
+        if (accession.totalWithdrawnCount != null || accession.totalWithdrawnWeight != null) {
```
Comment (karthikbtf) [inline src/main/kotlin/com/terraformation/backend/seedbank/db/MigrateWithdrawnTotals.kt:45]: probably unlikely .. but in the case of a re-run do we want to skip those that already have it set?

**Review by karthikbtf — APPROVED** [review]
looks good 👍 
guessing the UI will need to show these new values in accessions columns

**Comment by sgrimm** [comment]
Yep, https://github.com/terraware/terraware-web/pull/1122 is the frontend part of this.

File: src/main/kotlin/com/terraformation/backend/seedbank/db/AccessionStore.kt (hunk 1 above)
Comment (sgrimm) [reply src/main/kotlin/com/terraformation/backend/seedbank/db/AccessionStore.kt:910]: My thinking was that we'd want to abort the migration on exception, but you're right, I should make that explicit so the contract is well-defined.

File: src/main/kotlin/com/terraformation/backend/seedbank/db/MigrateWithdrawnTotals.kt (hunk 2 above)
Comment (sgrimm) [reply src/main/kotlin/com/terraformation/backend/seedbank/db/MigrateWithdrawnTotals.kt:45]: The search condition should do that, but I should make it clearer what's going on. Thanks.

**Pull request merged**